librz/core/cflag.c

```diff
@@ -0,0 +1,187 @@
+#include <rz_flag.h>
+#include <rz_util.h>
+#include <rz_cons.h>
+
+struct print_flag_t {
+	RzFlag *f;
+	PJ *pj;
+	bool in_range;
+	ut64 range_from;
+	ut64 range_to;
+	RzSpace *fs;
+	bool real;
+};
+
+static bool print_flag_name(RzFlagItem *fi, void *user) {
+	rz_cons_printf("%s\n", fi->name);
+	return true;
+}
+
+static bool print_flag_json(RzFlagItem *flag, void *user) {
+	struct print_flag_t *u = (struct print_flag_t *)user;
+	if (u->in_range && (flag->offset < u->range_from || flag->offset >= u->range_to)) {
+		return true;
+	}
+	pj_o(u->pj);
+	pj_ks(u->pj, "name", flag->name);
+	if (flag->name != flag->realname) {
+		pj_ks(u->pj, "realname", flag->realname);
+	}
+	pj_ki(u->pj, "size", flag->size);
+	if (flag->alias) {
+		pj_ks(u->pj, "alias", flag->alias);
+	} else {
+		pj_kn(u->pj, "offset", flag->offset);
+	}
+	if (flag->comment) {
+		pj_ks(u->pj, "comment", flag->comment);
+	}
+	pj_end(u->pj);
+	return true;
+}
+
+static bool print_flag_rizin(RzFlagItem *flag, void *user) {
+	struct print_flag_t *u = (struct print_flag_t *)user;
+	char *comment_b64 = NULL, *tmp = NULL;
+	if (u->in_range && (flag->offset < u->range_from || flag->offset >= u->range_to)) {
+		return true;
+	}
+	if (!u->fs || flag->space != u->fs) {
+		u->fs = flag->space;
+		rz_cons_printf("fs %s\n", u->fs ? u->fs->name : "*");
+	}
+	if (flag->comment && *flag->comment) {
+		comment_b64 = rz_base64_encode_dyn((const ut8 *)flag->comment, strlen(flag->comment));
+		// prefix the armored string with "base64:"
+		if (comment_b64) {
+			tmp = rz_str_newf("base64:%s", comment_b64);
+			free(comment_b64);
+			comment_b64 = tmp;
+		}
+	}
+	if (flag->alias) {
+		rz_cons_printf("fa %s %s\n", flag->name, flag->alias);
+		if (comment_b64) {
+			rz_cons_printf("\"fC %s %s\"\n",
+				flag->name, comment_b64 ? comment_b64 : "");
+		}
+	} else {
+		rz_cons_printf("f %s %" PFMT64d " 0x%08" PFMT64x " %s\n",
+			flag->name, flag->size, flag->offset,
+			comment_b64 ? comment_b64 : "");
+	}
+
+	free(comment_b64);
+	return true;
+}
+
+static bool print_flag_orig_name(RzFlagItem *flag, void *user) {
+	struct print_flag_t *u = (struct print_flag_t *)user;
+	if (u->in_range && (flag->offset < u->range_from || flag->offset >= u->range_to)) {
+		return true;
+	}
+	if (flag->alias) {
+		const char *n = u->real ? flag->realname : flag->name;
+		rz_cons_printf("%s %" PFMT64d " %s\n", flag->alias, flag->size, n);
+	} else {
+		const char *n = u->real ? flag->realname : (u->f->realnames ? flag->realname : flag->name);
+		rz_cons_printf("0x%08" PFMT64x " %" PFMT64d " %s\n", flag->offset, flag->size, n);
+	}
+	return true;
+}
+
+typedef struct {
+	RzTable *t;
+} FlagTableData;
+
+static bool __tableItemCallback(RzFlagItem *flag, void *user) {
+	FlagTableData *ftd = user;
+	if (!RZ_STR_ISEMPTY(flag->name)) {
+		RzTable *t = ftd->t;
+		const char *spaceName = (flag->space && flag->space->name) ? flag->space->name : "";
+		rz_table_add_rowf(t, "Xdss", flag->offset, flag->size, spaceName, flag->name);
```

Comment:
```suggestion
		rz_table_add_rowf(t, "Xxss", flag->offset, flag->size, spaceName, flag->name);
```

`Xxss` is used below.